core/genesis_write.go

```diff
@@ -192,7 +192,7 @@ func WriteGenesisBlock(tx kv.RwTx, genesis *types.Genesis, overridePragueTime *b
 	}
 
 	// set unwanted forks block to max number, so they are not activated
-	if newCfg.NormalcyBlock != nil && newCfg.NormalcyBlock.Cmp(big.NewInt(0)) != 0 && !honourChainspec {
+	if newCfg.NormalcyBlock != nil && newCfg.NormalcyBlock.Cmp(big.NewInt(0)) != 0 && genesis != nil && !genesis.HonourChainspec {
```

Comment:
I don't think we need the != 0 check here on normalcy as this will always be 0 for now.  Would keep it simpler to have `if normalcy and not honouring spec` as the logic